xbmc/platform/Curl.cpp

```diff
@@ -0,0 +1,663 @@
+#else
+  ec = make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_PROGRESSFUNCTION, cb);
+#endif
+  if (ec)
+    return ec;
+
+  return make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_NOPROGRESS, 0));
+}
+
+std::error_code CCurl::SetCallbacks(void* state,
+                                    ReadWriteCallback write,
+                                    ReadWriteCallback read,
+                                    HeaderCallback header)
+{
+  std::error_code ec;
+
+  ec = make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_WRITEDATA, state));
+  if (ec)
+    return ec;
+
+  ec = make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_WRITEFUNCTION, write));
+  if (ec)
+    return ec;
+
+  ec = make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_READDATA, state));
+  if (ec)
+    return ec;
+
+  ec = make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_READFUNCTION, read));
+  if (ec)
+    return ec;
+
+  ec = make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_HEADERDATA, state));
+  if (ec)
+    return ec;
+
+  ec = make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_HEADERFUNCTION, header));
+  if (ec)
+    return ec;
+
+  return make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_HEADER, 0));
+}
+
+std::error_code CCurl::SetUsernamePassword(const std::string& user, const std::string& pass)
+{
+  assert(!user.empty() && !pass.empty());
+
+  auto up = user + ":" + pass;
+  return make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_USERPWD, up.c_str()));
+}
+
+std::error_code CCurl::SetCookies(const std::string& cookies)
+{
+  return make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_COOKIE, cookies.c_str()));
+}
+
+std::error_code CCurl::SetAlias(const std::string& alias)
+{
+  if (m_session->m_alias)
+  {
+    curl_slist_free_all(m_session->m_alias);
+    m_session->m_alias = nullptr;
+  }
+
+  m_session->m_alias = curl_slist_append(m_session->m_alias, alias.c_str());
+
+  return make_error_code(
+      curl_easy_setopt(m_session->m_easy, CURLOPT_HTTP200ALIASES, m_session->m_alias));
+}
+
+std::error_code CCurl::SetUrl(const std::string& url)
+{
+  return make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_URL, url.c_str()));
+}
+
+std::error_code CCurl::SetPostData(const std::string& data)
+{
+  auto ec = make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_POST, 1));
+  if (ec)
+    return ec;
+
+  ec = make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_POSTFIELDSIZE, data.length()));
+  if (ec)
+    return ec;
+
+  return make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_POSTFIELDS, data.c_str()));
+}
+
+std::error_code CCurl::SetOrClearReferer(const std::string& referer)
+{
+  if (!referer.empty())
+    return make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_REFERER, referer.c_str()));
+
+  auto ec = make_error_code(curl_easy_setopt(m_session->m_easy, CURLOPT_REFERER, nullptr));
```

Comment:
`..., CURLOPT_REFERER, NULL));` to get it compiled for windows uwp

Reply:
Interesting. Didn't try building for uwp, so it handles C varargs differently?